src/snakemake/workflow.py

```diff
@@ -2277,16 +2278,31 @@ def userule(
         def decorate(maybe_ruleinfo):
             if from_module is not None:
                 try:
+                    modifier = name_modifier
                     module = self.modules[from_module]
                 except KeyError:
-                    raise WorkflowError(
-                        "Module {} has not been registered with 'module' statement before using it in 'use rule' statement.".format(
-                            from_module
+                    from inspect import currentframe
+
+                    if from_module in currentframe().f_back.f_globals:
+                        module = self.modules[
+                            currentframe().f_back.f_globals[from_module]
+                        ]
+                        if name_modifier is not None:
+                            if name_modifier.endswith("*"):
+                                modifier = f"{currentframe().f_back.f_globals[name_modifier[:-1]]}*"
+                            else:
+                                modifier = currentframe().f_back.f_globals[
+                                    name_modifier
+                                ]
+                    else:
+                        raise WorkflowError(
+                            "Module {} has not been registered with 'module' statement before using it in 'use rule' statement.".format(
+                                from_module
+                            )
```

Comment:
Can you add some explanatory comments here? also what kind of fallback this adds (I know now, but I will forget that within the next 2 months ;-))